### PR TITLE
Beef up ESQuery documentation.

### PR DESCRIPTION
**Pull request by esoergel**
Added some examples as documentation and wrote proper docs for things that felt like they needed it.  I kinda just wanted to see what it'd look like to have something fully documented.  I'm interested to see how much use people get from it.  This will be accessible both in the relevant files and on ReadTheDocs.
@anyone @czue

### CODE REVIEW

File: corehq/apps/es/__init__.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
+from .es_query import ESQuery, HQESQuery
```
Comment (dimagimon) [inline corehq/apps/es/__init__.py]: F401 'ESQuery' imported but unused

Comment (dimagimon) [inline corehq/apps/es/__init__.py]: F401 'HQESQuery' imported but unused

File: corehq/apps/es/__init__.py [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
+
 from . import filters
-from .es_query import ESQuery, ESQuerySet, HQESQuery
+from . import queries
```
Comment (dimagimon) [inline corehq/apps/es/__init__.py]: F401 'queries' imported but unused

File: corehq/apps/es/__init__.py [hunk 3]
```diff
@@ -1,11 +1,13 @@
+from .es_query import ESQuery, HQESQuery
```
Comment (dimagimon) [inline corehq/apps/es/__init__.py]: F401 'ESQuery' imported but unused

Comment (dimagimon) [inline corehq/apps/es/__init__.py]: F401 'HQESQuery' imported but unused

File: corehq/apps/es/__init__.py [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
+
 from . import filters
-from .es_query import ESQuery, ESQuerySet, HQESQuery
+from . import queries
```
Comment (dimagimon) [inline corehq/apps/es/__init__.py]: F401 'queries' imported but unused

**Comment by czue** [comment]
didn't read closely, but nice!

**Comment by czue** [comment]
(kicked build)

**Comment by snopoke** [comment]
kicked build again

**Comment by benrudolph** [comment]
sweet! es in hq has always been a bit of an enigma for me

**Pull request merged**